### PR TITLE
TTS Model

### PR DESCRIPTION
**Pull request by aamirchaudry29**
TTS Server is ran in conda environment (tts-server) -> ac2220/main.py

### CODE REVIEW

File: desktop/electron/main.ts [hunk 1]
```diff
@@ -101,6 +101,12 @@ app && app.whenReady().then(() => {
     }
   })
 
+  // handler for textToAudio function
```
Comment (acowusu) [inline desktop/electron/main.ts:104]: I think using api / IAPI  would be cleaner as it gives automatic typescript support and reduces boilerplate. You can see examples in routes.ts

File: desktop/electron/reportProcessing.ts [hunk 2]
```diff
@@ -14,11 +14,43 @@ import topics from "./mockData/topics.json";
...
+// Takes in an array of strings and returns an array of AudioInfo
+export async function textToAudio(textArray: string[]): Promise<AudioInfo[]> {
+
+    const audioInfoArray: AudioInfo[] = [];
+
+    for (const text of textArray) {
+      const postData = new FormData();
+      postData.append('script', text);
+
+      const response = await fetch('http://0.0.0.0:8888/tts/', {
```
Comment (acowusu) [inline desktop/electron/reportProcessing.ts:32]: This assumes the server is running on the user's local machine or the port is being port forwarded. Instead use https://iguana.alexo.uk/tts/

File: desktop/package.json [hunk 3]
```diff
@@ -91,7 +91,7 @@
     "@vitest/coverage-v8": "^1.2.2",
     "autoprefixer": "^10.4.16",
     "better-sqlite3": "^8.6.0",
-    "electron": "^26.1.0",
+    "electron": "^26.6.9",
```
Comment (acowusu) [inline desktop/package.json:94]: What is this change for?

**Comment by aamirchaudry29** [comment]
Soz just ignore this pull ill make changes based off what we spoke about
today

**Comment by aamirchaudry29** [comment]
Close

**Pull request closed without merging**